src/http_server.cc

```diff
@@ -1661,6 +1661,17 @@ HTTPAPIServer::HandleTrace(evhtp_request_t* req, const std::string& model_name)
   int32_t count;
   uint32_t log_frequency;
   std::string filepath;
+  if (!model_name.empty()) {
+    bool ready = false;
+    auto err = TRITONSERVER_ServerModelIsReady(
+        server_.get(), model_name.c_str(), -1 /* model version */, &ready);
+    if (!ready || err != nullptr) {
+      HTTP_RESPOND_IF_ERR(
+          req, TRITONSERVER_ErrorNew(
+                   TRITONSERVER_ERROR_INVALID_ARG,
+                   ("Request for unknown model : " + model_name).c_str()));
+    }
```

Comment:
I believe this is leaking `err` without deleting it. You can probably do something like:
```cpp
bool ready = false;
HTTP_RESPOND_IF_ERR(req, TRITONSERVER_ServerModelIsReady(..., &ready));
if (!ready) {
      HTTP_RESPOND_IF_ERR(
          req, TRITONSERVER_ErrorNew(
                   TRITONSERVER_ERROR_INVALID_ARG,
                   ("Request for unknown model : " + model_name).c_str()));
}
```
Similarly for GRPC server, don't want to overwrite a non-null error without deleting it.

Reply:
Good catch, thanks! By that same token, would [this](https://github.com/triton-inference-server/server/blob/8a823b2381f12f97219404543ad55b1437a62958/src/http_server.cc#L1283-L1293C43) also be a memory leak since this newly declared `err` variable could potentially leave scope without being deleted?

Reply:
It gets cleaned up [here](https://github.com/triton-inference-server/server/blob/8a823b2381f12f97219404543ad55b1437a62958/src/http_server.cc#L1300)

Reply:
I think that cleans up [this](https://github.com/triton-inference-server/server/blob/8a823b2381f12f97219404543ad55b1437a62958/src/http_server.cc#L1240C3-L1240C37) variable `err` , declared earlier. The variable of the same name that's declared on 1283 looks to go out of scope unhandled on line 1295, no?

Reply:
Ah you're right, 1283 appears to shadow the outer `err` var, this is probably leaking as well. As discussed offline, you should quickly verify this with valgrind, and probably replace `auto err = ...` on 1283 with just `err = ...` to use the outer scope variable.

Reply:
Confirmed with valgrind that it could indeed cause a memory leak. Adding the fix to the PR.